engines/wage/sound.cpp

```diff
@@ -122,6 +122,10 @@ static void soundTimer(void *refCon) {
 	if (engine->_world->_player->_currentScene != scene)
 		return;
 
+	if (!engine->_soundQueue.empty() && engine->_soundQueue.front() < g_system->getMillis()) {
```

Comment:
This does not look right to me. E.g. if, let's say due to some OS delay, your callback is called 1 ms later, you will kill the sound.

What you need to do in order to fix it properly, is clear the sound queue on the scene change.